### DISCUSSION TITLE
2024-02-27 题库讨论

### DISCUSSION THREAD

**Discussion by xbn2002** [MlgmXyysd/Xiaomi-BootLoader-Questionnaire#89] [General]
> ### 申请前必读
> 
> 1. Bootloader解锁会导致设备面临各类安全风险，并对设备造成不可预估的影响，申请解锁代表您已经充分了解相关风险
> 
> 2. 解锁后请勿将设备用于非正常用途，若检测到账号异常行为，可能导致账号被封禁
> 
> 3. 本项目仅适合正在使用小米澎湃OS且设备处于上锁状态的用户报名，如果您的设备暂未升级到小米澎湃OS，您可以继续通过原有途径获取BL锁解锁权限
> 
> 4. 为确保答题的公平性，提升答题体验，每周将安排四场答题时间：每周二上午8:30 - 8:45、10:45 - 11:00，每周四上午8:30 - 8:45、10:45 - 11:00，您可以任选其中一个时段参与，每次参与需间隔7天，还请合理安排答题时间（温馨提示：每场答题的最迟进场时间为该场次答题开始后的10分钟）
> 
> 5. **答题成绩每周一晚上23:59失效**，如答题通过，请务必在周一晚上23:59前完成资格申请，避免成绩失效影响申请
> 
> 6. 报名成功不代表审核通过，解锁权限获取以最终审核结果为准；通过社区申请获得解锁权限后，若您绑定设备时该设备系统为小米澎湃OS，绑定成功后解锁等待期为72小时，请您知悉

- 答题链接：https://m.beehive.miui.com/wlH2yKwBvy_Iu85HjlYFjg/

> ### BL解锁申请资格答题测试
> 
> 【填空题说明】填空题采用人工阅卷方式，人工阅卷预计在【下一答题周期开始前】完成。①您提交问卷时的实时成绩仅为【选择题得分】；②人工阅卷后，如您的【选择题+填空题总分】未达到分数线时，您的成绩将仅展示【选择题得分】；③人工阅卷后，如您的【选择题+填空题总分】达到分数线时，您的成绩将展示为【选择题+填空题总分】
> 【无法展示成绩的原因】①请务必通过“小米社区App-内测中心-BL解锁申请-去答题”进入本次答题界面并完成答题，未使用小米社区App进行答题将视为无效答题，无法展示成绩；②等级未达5段，无法展示成绩；③答题超时提交，无法展示成绩
> 请您充分知晓解锁和刷机存在的风险，并在可承担该风险的情况下申请Bootloader解锁
> 请您填写相关问题，为避免成绩无效，建议作答时间不少于【5分钟】且不超过【15分钟】，请合理安排答题时间。本期答题得分高于85分（不含85分）即可通过，感谢您的支持！

**Comment by xbn2002** [comment]
08:30-08:45
- 1546436 **以下关于《小米账号使用协议》和《BL解锁申请资格答题测试》的相关要求，说法正确的有** 【多选题】
  - 5071889 填空题务必不要复制粘贴其他网站的答案，答案雷同或相似度高，将视为无效问卷，无法获得分数
    - ✔️ 选中。根据问卷填空题页面提示，如出现使用AI生成、复制粘贴、雷同试卷等情况，将视为无效作答，无法通过审核。
  - 5071890 如果小米发现您并非该账号初始申请注册人，小米有权在未经通知的情况下终止您使用该账号
    - ✔️ 选中。根据《小米账号使用协议》，任何情况下，您都无权使用由他人注册的账号。
  - 5071891 当您获得BL解锁资格后，通过其他途径参与本答题测试，会根据行为严重程度，面临取消资格、账号终止使用等处罚
    - ✔️ 选中。根据《BL解锁申请资格答题测试》的相关要求，您只能通过「小米社区」APP 参与答题。
  - 5071892 当您小米社区等级未达到5段及以上时，通过其他途径参与本答题测试，会根据行为严重程度，面临账号终止使用等处罚
    - ✔️ 选中。本选项的表述是对上一选项的补充。
- 1546437 **已满足BL解锁申请条件，但申请后被拒绝的原因，下列说法正确的有** 【多选题】
  - 5071893 在社交媒介恶意诋毁、谩骂、造谣等
    - ✔️ 选中。您不得在社交媒介对小米恶意诋毁、谩骂、造谣等。
  - 5071894 出现被判定为作弊的行为
    - ✔️ 选中。根据解锁申请详细说明，“代答”等账号异常行为属于违规，解锁申请不予通过，且账号将被封禁。
  - 5071895 不填写申请理由、或使用AI生成申请理由、或申请理由出现过辱骂威胁等内容
    - ✔️ 选中。您必须在申请时认真填写真实的申请理由。
  - 5071896 其他违反小米账号使用协议的相关情形
    - ✔️ 选中。您不得违反小米账号使用协议。
  - 5071897 确认无任何违规情形，被拒绝后可尝试再次申请
    - ✔️ 选中。如您无上述违规情形，在申请被拒绝后，可在 7 天后重新答题并申请。
- 1546438 **BL解锁面向发烧友和开发者开放，因此在资格筛选时，会考察是否符合发烧友的特质。下面关于发烧友的特质，说法正确的有** 【多选题】
  - 5071898 发烧友指的是对某一领域极度热爱，并对此领域的产品进行深入研究
    - ✔️ 选中。公告类型题，对上一题的正确选项做出了解释。
  - 5071899 发烧友具备探究精神，因此在遇到不了解的问题时，会在合理范围内主动探寻答案，但不会寻求不正当“捷径”或采用违规方式
    - ✔️ 选中。
  - 5071900 发烧友具备坚韧不拔的品质，因此在遇到失败或挫折时，会不断尝试和努力，但不会去造谣、诋毁
    - ✔️ 选中。
  - 5071901 发烧友具备学习精神，因此在遇到新鲜事物或新知识时，会乐于学习和钻研，不会肆意谩骂
    - ✔️ 选中。
  - 5071902 发烧友具备分享精神，因此在有一些好的知识或经验时，会积极分享和交流，不会去利用“资源”去“谋利”
    - ✔️ 选中。
- 1546439 **BL解锁危害多，BL解锁可能造成各类App无法正常使用、个人数据信息泄露、手机变砖、资产损失等情况，因此强烈建议您非必要情况不申请BL解锁。请问您是否已经知悉以上风险和危害，并愿意承担以上全部风险和危害？** 【单选题】
  - 5071903 本人愿意承担以上全部风险和危害，接受各类App无法正常使用、个人数据信息泄露、手机变砖、资产损失等情况
    - ✔️ 选中。免责声明，不做解释。
  - 5071904 本人不愿意承担以上全部风险和危害，立即关闭界面，停止答题和申请
    - ❌ 不选。
- 1546440 **请简述线刷和卡刷的步骤** （此题为人工阅卷，如出现使用AI生成、复制粘贴、雷同试卷等情况，将视为无效作答，无法通过审核） 【多项填空题】
  - 328268 线刷
    - ✔️ 至少需要输入 20 个字符。从小米官方服务器下载 TGZ 格式的全量包，将手机 BootLoader 解锁，并重启至 BootLoader 模式（手机中显示 FASTBOOT），连接电脑执行 flash_all 脚本刷入。
  - 328269 卡刷
    - ✔️ 至少需要输入 20 个字符。从小米官方服务器下载 ZIP 格式的全量包，使用「设置 - 系统更新 - 手动选择安装包」或 Recovery 模式刷入。
- 1546448 **以下说法不正确的有** 【多选题】
  - 5071929 某二叉树有5个叶结点，其权值分别为10、12、16、21、30，则其最小的带权路径长度是200
    - ❌ 不选。带权路径长度（Weighted Path Length, WPL）是一种用于衡量树形结构的编码长度的方法，通常应用在哈夫曼树（Huffman Tree）中。给定权值 10、12、16、21、30，要使带权路径长度最小，可构造如下哈夫曼树：
    <div align="center"><img src = "https://github.com/MlgmXyysd/Xiaomi-BootLoader-Questionnaire/assets/27143040/fb3ed2dc-84e0-4818-a33c-63e02814aa39" /></div>
    $$WPL=(16+21+30)*2+(10+12)*3=200$$
  - 5071930 计算机主存地址为32位，按字节编址，Cache数据区大小和主存块大小均为32KB，采用直接映射方式和回写策略，则Cache行的位数至少是274
    - ✔️ 选中。本题正确答案应为 275。具体解题过程见[视频解析](https://www.acfun.cn/v/ac35280659)。
  - 5071931 线性表若采用链式存储结构时，则内存中可用存储单元的地址必须连续
    - ✔️ 选中。线性表采用链式存储结构时，内存中可用存储单元的地址连续或不连续都可以。
  - 5071933 线刷可以用来降级回到之前的系统版本、彻底删除设备数据等
    - ❌ 不选。该描述是正确的。
  - 5071949 用三叉链表作为二叉树的存储结构，当二叉树有n个结点时，有n+2个空指针
    - ❌ 不选。用三叉链表作为二叉树的存储结构，当二叉树有 $n$ 个结点时，应该有 $n+2$ 个空指针。
- 1546441 **请简述手机BL解锁后可能产生的风险** （此题为人工阅卷，如出现使用AI生成、复制粘贴、雷同试卷等情况，将视为无效作答，无法通过审核） 【多项填空题】
  - 328270 风险1
    - ✔️ 至少需要输入 20 个字符。解锁后设备无任何安全保障，易受恶意软件攻击，个人隐私数据可能会因此泄露，并可能影响后续的系统升级。获取 root 权限后，您的设备将面临被恶意格机等风险，并可能会在一定程度上导致设备的软硬件损毁。
  - 328271 风险2
    - ✔️ 至少需要输入 20 个字符。解锁后设备查找功能无法使用，因此当您遗失设备时，将面临更高的经济损失风险。设备保修和购买的延保等权益将会丢失，若出现无论是否系人为导致的质量问题，均不能享受免费售后服务。
  - 328272 风险3
    - ✔️ 至少需要输入 20 个字符。解锁后执行特定刷机操作可能会导致 TEE 永久熔断，从而导致部分对系统安全性依赖高的功能和服务失效，并可能造成各类 App（特别是金融类、支付类等对于安全性有较高要求的 App）无法正常使用。
  - 328427 风险4
    - ✔️ 至少需要输入 20 个字符。解锁后刷入不相容的包或模块将可能会导致手机变砖。当设备无法进入 fastboot 模式，或者线刷仍然无法解决问题时，您必须前往小米官方授权维修服务中心使用深刷模式救砖。若解锁后设备用于非正常用途，账号还将面临被封禁等风险。
- 1546442 **输入fastboot devices命令后，可能输出的结果是** 【多选题】
  - 5071905 $ wget https://xxx.xxx/xxx/xxx/xxx.img
    - ❌ 不选。该内容为用户手动输入的指令。
  - 5071906 MBR: verify success!
    - ❌ 不选。该内容为引导加载程序的日志。
  - 5071907 remount failed
    - ❌ 不选。该内容为挂载（mount）命令的报错。
  - 5071908 XMABC100-01 fastboot
    - ✔️ 选中。fastboot 子命令 devices 的输出格式为空（即无设备连接）或「<序列号> fastboot」（如有多个设备同时连接，则每行会显示一个设备）。
- 1546443 **以下FASTBOOT基本命令，对应正确的有** 【多选题】
  - 5071909 fastboot oem device-info：输出该设备的所有信息
    - ❌ 不选。子命令 oem 为制造商自定义命令。对于 Xiaomi/Redmi 手机，device-info 命令将仅能输出如下设备状态消息：Verity mode（完整性校验状态），刷入非官方镜像需要禁用，为 false 时表示已禁用完整性校验；Device unlocked（BootLoader 解锁状态），为 true 时表示已解锁；Device critical unlocked（关键分区解锁状态），为 true 时表示 BootLoader 已完整解锁；Charger screen enabled（关机充电屏幕状态）。
  - 5071910 fastboot oem poweroff：清除手机中所有数据后关机
    - ❌ 不选。子命令 oem 为制造商自定义命令。对于 Xiaomi/Redmi 手机，poweroff 命令只会关机，不会清除数据。
  - 5071911 fastboot reboot：重启手机
    - ✔️ 选中。子命令 reboot 用于重启设备。
  - 5071912 fastboot erase system：擦除系统分区
    - ✔️ 选中。子命令 erase 用于擦除指定分区的所有数据。system 为 Android 系统分区。
  - 5071913 fastboot delete：擦除所有分区
    - ❌ 不选。fastboot 不存在 delete 子命令。同时，亦不存在能直接擦除所有分区的命令。
- 1546444 **MTK刷机的报错，原因不匹配的有** 【多选题】
  - 5071914 S_UNSUPPORTED_OPERATION：已存在同名的程序
    - ✔️ 选中。该报错意味着刷机引导文件 Scatter.txt 与设备不匹配。
  - 5071915 S_DA_EMMC_FLASH_NOT_FOUND：手机的内存己经损坏
    - ✔️ 选中。该报错意味着刷机引导文件 Scatter.txt 与设备不匹配。
  - 5071916 S_FT_NEED_DOWNLOAD_ALL_FAIL：目标ROM文件和手机系统差距过大
    - ❌ 不选。该报错意味着目标 ROM 文件和手机的系统差距过大，通常在单刷某个分区时会出现。
  - 5071917 S_BROM_CMD_STARTCMD_FAIL：没有加载好刷机包文件
    - ❌ 不选。该报错意味着没有加载好刷机包的文件或驱动没装好。
  - 5071918 S_CHIP_TYPE_NOT_MATCH：文件类型不匹配
    - ✔️ 选中。该报错意味着刷机包适配的芯片型号与设备不匹配。
- 1546447 **请截图本页面，并上传截图** 【文件题】
  - 0 未正确上传该页面截图或伪造截图，将视为无效答卷，无法通过审核
    - ✔️ 至少需要上传 1 个文件。请使用「小米社区」APP 答题并截图上传。
- 1546445 **以下说法正确的有** 【多选题】
  - 5071919 设置界地址保护是分区分配内存管理方式的主要保护措施
    - ✔️ 选中。该描述是正确的。
  - 5071920 将长度分别为m,n的两个单链表合并为一个单链表的时间复杂度为O(m*n)
    - ❌ 不选。获得短链表尾指针需要 $O(min(m,n))$ 的时间，随后只需要将长的单链表接在后面即可。故将两个单链表合并为一个单链表的时间复杂度为 $O(min(m,n))$。
  - 5071921 一棵完全二叉树第六层有8个叶结点，则该二叉树的第5层结点个数为32个（根为第一层）
    - ❌ 不选。根据下表可知，该完全二叉树在第 5 层一共有 16 个节点，不是 32 个。
    <table align="center">
        <tr align="center">
            <th>层</th>
            <th>节点数</th>
        </tr>
        <tr align="center">
            <td>1</td>
            <td>1</td>
        </tr>
        <tr align="center">
            <td>2</td>
            <td>2</td>
        </tr>
        <tr align="center">
            <td>3</td>
            <td>4</td>
        </tr>
        <tr align="center">
            <td>4</td>
            <td>8</td>
        </tr>
        <tr align="center">
            <td>5</td>
            <td>16</td>
        </tr>
        <tr align="center">
            <td>6</td>
            <td>32</td>
        </tr>
        <tr align="center">
            <td>7</td>
            <td>48</td>
        </tr>
    </table>
  - 5071922 输入“adb logcat”后，出现“waiting for device”信息，可能是由于设备没有连接计算机
    - ✔️ 选中。该信息的含义为 【等待 device 模式的设备连接】，可能出现的原因有以下几点：1）驱动未安装或出现异常，计算机无法识别设备；2）设备没有物理连接到计算机，可检查接口是否松动或数据线是否正常；3）设备不在 device 模式，或未授权计算机；4）设备没有启用「USB 调试」功能。
  - 5071923 若一个栈的入栈序列为1、2、3、...、N，输出序列的第一个元素是i，则无法确定第j个输出的元素
    - ✔️ 选中。由于可以交替进行入栈、出栈操作，并不一定要将栈内元素全部出栈后才能入栈，也不一定要将序列中的元素全部进栈后才能出栈，所以无法确定第 $j$ 个输出的元素。
- 1547177 **请输入FASTBOOT模式下输出防回滚机制版本号的命令** 【填空题】
  - 0 
    - ✔️ 至少需要输入 20 个字符。正确的输出小米防回滚机制（Xiaomi Anti-Rollback）版本号命令为 fastboot getvar anti。
- 1546446 **以下说法正确的有** 【多选题】
  - 5071925 若采用三元组表存储结构存储稀疏矩阵M，则除三元组外，还需要保存矩阵M的行数和列数
    - ✔️ 选中。三元组是（非零元素的值 $value$，所在行 $i$，所在列 $j$），无法推定矩阵 $M$ 的边界上还存在多少零元素。因此要想确定整个矩阵，还需要保存矩阵 $M$ 的行数和列数。
  - 5071926 刷机过程中，刷入的系统版本与原系统差异较大时，通常建议执行二清
    - ❌ 不选。当刷入系统差异较大时，如果不完整清除数据，可能会无法开机，所以二清是不够的，往往需要四清。二清指的是用户数据（Userdata 或 Data）和系统缓存（Cache）；四清指的是用户数据（Userdata 或 Data）、加密元数据（Metadata）、系统缓存（Cache）以及账户锁（Frp）。
  - 5071927 BootLoader是在Android设备启动时最先加载到RAM中的程序，它的主要作用是把系统OS拉起来并运行
    - ❌ 不选。根据 Android 文档，引导加载器（BootLoader）负责在设备上启动内核，而不是系统。
  - 5071928 若计算机中的地址总线的宽度为24位，则最多允许直接访问主存储器的8MB物理空间
    - ❌ 不选。如果地址总线的宽度为 24 位，那么它可以产生 2 ^ 24 个不同的地址。每个地址对应内存中的 1 个字节，所以最多允许直接寻址的主存储器物理空间大小为 2 ^ 24 = 16777216 B，即 16 MB。
  - 5071948 9008模式下的刷机比fastboot模式下的刷机更加彻底
    - ✔️ 选中。Qualcomm Emergency Download（EDL 或 9008）模式刷机更为底层，能够写入一些在 fastboot 模式下无法写入的分区或数据。
- 1547178 **如何退出FASTBOOT模式？** 【填空题】
  - 0 
    - ✔️ 至少需要输入 20 个字符。
    （1）在 FASTBOOT 模式下长按电源键约 7-15 秒重启进入系统。
    （2）通过 "fastboot reboot" 及 "fastboot oem poweroff" 命令重启进入系统或拔线关机。
    （3）部分机型在 FASTBOOT 模式下不插入数据线，5分钟到半小时内会自动重启或关机。
- 1546450 **请简述对“小米BL解锁申请”的看法** 【填空题】
  - 0 此题为人工阅卷，如出现使用AI生成、复制粘贴、雷同试卷等情况，将视为无效作答，无法通过审核
    - ✔️ 至少需要输入 20 个字符。「？」（本题自行填写，言之有理即可）
- 1546451 **如您在其他技术交流平台有较多贡献（如：GitHub等），可将相关信息进行填写，并在下一题上传相关截图凭证** 【填空题】
  - 0 非必填，此题的有效填写会提升审核通过率
    - ✔️ 非必填。「？」
- 1546452 **请将上一题填写的其他技术交流平台贡献信息，上传相关截图凭证，以供审核使用** 【文件题】
  - 0 非必填，此题的有效填写会提升审核通过率
    - ✔️ 非必填。如您上一题填写了相关内容，请于本题上传截图凭证。
- 1547175 **【温馨提示】将下面的提示文案内容完整输入至答题框内** 【填空题】
  - 0 本人知悉需要在倒计时结束前提交本问卷，超时提交将无法展示分数。
    - ✔️ 至少需要输入 20 个字符。本人知悉需要在倒计时结束前提交本问卷，超时提交将无法展示分数。

**Reply by xbn2002** [reply]
10:45-11:00
- 1546436 **以下关于《小米账号使用协议》和《BL解锁申请资格答题测试》的相关要求，说法正确的有** 【多选题】
  - 5071889 填空题务必不要复制粘贴其他网站的答案，答案雷同或相似度高，将视为无效问卷，无法获得分数
    - ✔️ 选中。根据问卷填空题页面提示，如出现使用AI生成、复制粘贴、雷同试卷等情况，将视为无效作答，无法通过审核。
  - 5071890 如果小米发现您并非该账号初始申请注册人，小米有权在未经通知的情况下终止您使用该账号
    - ✔️ 选中。根据《小米账号使用协议》，任何情况下，您都无权使用由他人注册的账号。
  - 5071891 当您获得BL解锁资格后，通过其他途径参与本答题测试，会根据行为严重程度，面临取消资格、账号终止使用等处罚
    - ✔️ 选中。根据《BL解锁申请资格答题测试》的相关要求，您只能通过「小米社区」APP 参与答题。
  - 5071892 当您小米社区等级未达到5段及以上时，通过其他途径参与本答题测试，会根据行为严重程度，面临账号终止使用等处罚
    - ✔️ 选中。本选项的表述是对上一选项的补充。
- 1546437 **已满足BL解锁申请条件，但申请后被拒绝的原因，下列说法正确的有** 【多选题】
  - 5071893 在社交媒介恶意诋毁、谩骂、造谣等
    - ✔️ 选中。您不得在社交媒介对小米恶意诋毁、谩骂、造谣等。
  - 5071894 出现被判定为作弊的行为
    - ✔️ 选中。根据解锁申请详细说明，“代答”等账号异常行为属于违规，解锁申请不予通过，且账号将被封禁。
  - 5071895 不填写申请理由、或使用AI生成申请理由、或申请理由出现过辱骂威胁等内容
    - ✔️ 选中。您必须在申请时认真填写真实的申请理由。
  - 5071896 其他违反小米账号使用协议的相关情形
    - ✔️ 选中。您不得违反小米账号使用协议。
  - 5071897 确认无任何违规情形，被拒绝后可尝试再次申请
    - ✔️ 选中。如您无上述违规情形，在申请被拒绝后，可在 7 天后重新答题并申请。
- 1546438 **BL解锁面向发烧友和开发者开放，因此在资格筛选时，会考察是否符合发烧友的特质。下面关于发烧友的特质，说法正确的有** 【多选题】
  - 5071898 发烧友指的是对某一领域极度热爱，并对此领域的产品进行深入研究
    - ✔️ 选中。公告类型题，对上一题的正确选项做出了解释。
  - 5071899 发烧友具备探究精神，因此在遇到不了解的问题时，会在合理范围内主动探寻答案，但不会寻求不正当“捷径”或采用违规方式
    - ✔️ 选中。
  - 5071900 发烧友具备坚韧不拔的品质，因此在遇到失败或挫折时，会不断尝试和努力，但不会去造谣、诋毁
    - ✔️ 选中。
  - 5071901 发烧友具备学习精神，因此在遇到新鲜事物或新知识时，会乐于学习和钻研，不会肆意谩骂
    - ✔️ 选中。
  - 5071902 发烧友具备分享精神，因此在有一些好的知识或经验时，会积极分享和交流，不会去利用“资源”去“谋利”
    - ✔️ 选中。
- 1546439 **BL解锁危害多，BL解锁可能造成各类App无法正常使用、个人数据信息泄露、手机变砖、资产损失等情况，因此强烈建议您非必要情况不申请BL解锁。请问您是否已经知悉以上风险和危害，并愿意承担以上全部风险和危害？** 【单选题】
  - 5071903 本人愿意承担以上全部风险和危害，接受各类App无法正常使用、个人数据信息泄露、手机变砖、资产损失等情况
    - ✔️ 选中。免责声明，不做解释。
  - 5071904 本人不愿意承担以上全部风险和危害，立即关闭界面，停止答题和申请
    - ❌ 不选。
- 1546441 **请简述手机BL解锁后可能产生的风险** （此题为人工阅卷，如出现使用AI生成、复制粘贴、雷同试卷等情况，将视为无效作答，无法通过审核） 【多项填空题】
  - 328270 风险1
    - ✔️ 至少需要输入 20 个字符。解锁后设备无任何安全保障，易受恶意软件攻击，个人隐私数据可能会因此泄露，并可能影响后续的系统升级。获取 root 权限后，您的设备将面临被恶意格机等风险，并可能会在一定程度上导致设备的软硬件损毁。
  - 328271 风险2
    - ✔️ 至少需要输入 20 个字符。解锁后设备查找功能无法使用，因此当您遗失设备时，将面临更高的经济损失风险。设备保修和购买的延保等权益将会丢失，若出现无论是否系人为导致的质量问题，均不能享受免费售后服务。
  - 328272 风险3
    - ✔️ 至少需要输入 20 个字符。解锁后执行特定刷机操作可能会导致 TEE 永久熔断，从而导致部分对系统安全性依赖高的功能和服务失效，并可能造成各类 App（特别是金融类、支付类等对于安全性有较高要求的 App）无法正常使用。
  - 328427 风险4
    - ✔️ 至少需要输入 20 个字符。刷入不相容的包或模块将可能会导致手机变砖。当设备无法进入 fastboot 模式，或者线刷仍然无法解决问题时，您必须前往小米官方授权维修服务中心使用深刷模式救砖。若解锁后设备用于非正常用途，账号还将面临被封禁等风险。
- 1546442 **输入fastboot devices命令后，可能输出的结果是** 【多选题】
  - 5071905 $ wget https://xxx.xxx/xxx/xxx/xxx.img
    - ❌ 不选。该内容为用户手动输入的指令。
  - 5071906 MBR: verify success!
    - ❌ 不选。该内容为引导加载程序的日志。
  - 5071907 remount failed
    - ❌ 不选。该内容为挂载（mount）命令的报错。
  - 5071908 XMABC100-01 fastboot
    - ✔️ 选中。fastboot 子命令 devices 的输出格式为空（即无设备连接）或「<序列号> fastboot」（如有多个设备同时连接，则每行会显示一个设备）。
- 1546440 **请简述线刷和卡刷的步骤** （此题为人工阅卷，如出现使用AI生成、复制粘贴、雷同试卷等情况，将视为无效作答，无法通过审核） 【多项填空题】
  - 328268 线刷
    - ✔️ 至少需要输入 20 个字符。从小米官方服务器下载 TGZ 格式的全量包，将手机 BootLoader 解锁，并重启至 BootLoader 模式（手机中显示 FASTBOOT），连接电脑执行 flash_all 脚本刷入。
  - 328269 卡刷
    - ✔️ 至少需要输入 20 个字符。从小米官方服务器下载 ZIP 格式的全量包，使用「设置 - 系统更新 - 手动选择安装包」或 Recovery 模式刷入。
- 1546448 **以下说法不正确的有** 【多选题】
  - 5071929 刷机过程中如果变砖，不可以通过9008的模式刷机救回MTK平台的机器
    - ❌ 不选。仅高通（Qualcomm）平台的手机可以通过官方售后使用 Qualcomm Emergency Download（EDL, 9008 模式）刷机。联发科（MediaTek, MTK）平台的机器需使用 MediaTek Preloader。
  - 5071930 关机状态下长按电源键和音量-键可以进入FASTBOOT模式
    - ❌ 不选。Xiaomi/Redmi 手机进入 fastboot 的组合键一般是「电源键」和「音量减键」。
  - 5071931 线性表若采用链式存储结构时，则内存中可用存储单元的地址必须连续
    - ✔️ 选中。线性表采用链式存储结构时，内存中可用存储单元的地址连续或不连续都可以。
  - 5071933 线刷可以用来降级回到之前的系统版本、彻底删除设备数据等
    - ❌ 不选。该描述是正确的。
  - 5071949 线刷开发版ROM不需要解锁Bootloader
    - ✔️ 选中。线刷前必须先解锁 BootLoader。
- 1546444 **MTK刷机的报错，原因不匹配的有** 【多选题】
  - 5071914 S_UNSUPPORTED_OPERATION：已存在同名的程序
    - ✔️ 选中。该报错意味着刷机引导文件 Scatter.txt 与设备不匹配。
  - 5071915 S_DA_EMMC_FLASH_NOT_FOUND：手机的内存己经损坏
    - ✔️ 选中。该报错意味着刷机引导文件 Scatter.txt 与设备不匹配。
  - 5071916 S_FT_NEED_DOWNLOAD_ALL_FAIL：目标ROM文件和手机系统差距过大
    - ❌ 不选。该报错意味着目标 ROM 文件和手机的系统差距过大，通常在单刷某个分区时会出现。
  - 5071917 S_BROM_CMD_STARTCMD_FAIL：没有加载好刷机包文件
    - ❌ 不选。该报错意味着没有加载好刷机包的文件或驱动没装好。
  - 5071918 S_CHIP_TYPE_NOT_MATCH：文件类型不匹配
    - ✔️ 选中。该报错意味着刷机包适配的芯片型号与设备不匹配。
- 1546447 **请截图本页面，并上传截图** 【文件题】
  - 0 未正确上传该页面截图或伪造截图，将视为无效答卷，无法通过审核
    - ✔️ 至少需要上传 1 个文件。请使用「小米社区」APP 答题并截图上传。
- 1546445 **以下说法正确的有** 【多选题】
  - 5071919 在正常情况下，所有非system分区的APP运行时的UID都是一样的
    - ❌ 不选。Android 系统会给每个非系统应用（untrusted_app）分配不同的 UID。
  - 5071920 将长度分别为m,n的两个单链表合并为一个单链表的时间复杂度为O(m+n)
    - ❌ 不选。获得短链表尾指针需要 $O(min(m,n))$ 的时间，随后只需要将长的单链表接在后面即可。故将两个单链表合并为一个单链表的时间复杂度为 $O(min(m,n))$。
  - 5071921 算法可以没有输入，但必须要有输出
    - ✔️ 选中。该描述是正确的。
  - 5071922 输入“adb logcat”后，出现“waiting for device”信息，可能是由于设备没有连接计算机
    - ✔️ 选中。该信息的含义为 【等待 device 模式的设备连接】，可能出现的原因有以下几点：1）驱动未安装或出现异常，计算机无法识别设备；2）设备没有物理连接到计算机，可检查接口是否松动或数据线是否正常；3）设备不在 device 模式，或未授权计算机；4）设备没有启用「USB 调试」功能。
  - 5071923 若一个栈的入栈序列为1、2、3、...、N，输出序列的第一个元素是i，则无法确定第j个输出的元素
    - ✔️ 选中。由于可以交替进行入栈、出栈操作，并不一定要将栈内元素全部出栈后才能入栈，也不一定要将序列中的元素全部进栈后才能出栈，所以无法确定第 $j$ 个输出的元素。
- 1547177 **请写出“擦除系统分区”的FASTBOOT的命令** 【填空题】
  - 0 
    - ✔️ 至少需要输入 20 个字符。正确的擦除系统（system）分区的命令为 fastboot erase system。
- 1546446 **以下说法正确的有** 【多选题】
  - 5071925 擦除cache分区数据，个人数据通常不会丢失
    - ✔️ 选中。擦除缓存（cache）分区数据，通常不会涉及个人数据（Userdata 和 Metadata）。
  - 5071926 刷机过程中，刷入的系统版本与原系统差异较大时，通常建议执行二清
    - ❌ 不选。当刷入系统差异较大时，如果不完整清除数据，可能会无法开机，所以二清是不够的，往往需要四清。二清指的是用户数据（Userdata 或 Data）和系统缓存（Cache）；四清指的是用户数据（Userdata 或 Data）、加密元数据（Metadata）、系统缓存（Cache）以及账户锁（Frp）。
  - 5071927 BootLoader是在Android设备启动时最先加载到RAM中的程序，它的主要作用是把系统OS拉起来并运行
    - ❌ 不选。根据 Android 文档，引导加载器（BootLoader）负责在设备上启动内核，而不是系统。
  - 5071928 网络配置后不能进入Internet服务，ping 127.0.0.1是通的，ping 网关不通，原因可能是网关没有正确配置
    - ✔️ 选中。该问题的原因可能是网关没有正确配置，或者存在网络连接或路由问题。
  - 5071948 9008模式下的刷机比fastboot模式下的刷机更加彻底
    - ✔️ 选中。Qualcomm Emergency Download（EDL 或 9008）模式刷机更为底层，能够写入一些在 fastboot 模式下无法写入的分区或数据。
- 1547178 **如何退出FASTBOOT模式？** 【填空题】
  - 0 
    - ✔️ 至少需要输入 20 个字符。
    （1）在 FASTBOOT 模式下长按电源键约 7-15 秒重启进入系统。
    （2）通过 "fastboot reboot" 及 "fastboot oem poweroff" 命令重启进入系统或拔线关机。
    （3）部分机型在 FASTBOOT 模式下不插入数据线，5分钟到半小时内会自动重启或关机。
- 1546450 **请简述对“小米BL解锁申请”的看法** 【填空题】
  - 0 此题为人工阅卷，如出现使用AI生成、复制粘贴、雷同试卷等情况，将视为无效作答，无法通过审核
    - ✔️ 至少需要输入 20 个字符。「？」（本题自行填写，言之有理即可）
- 1546451 **如您在其他技术交流平台有较多贡献（如：GitHub等），可将相关信息进行填写，并在下一题上传相关截图凭证** 【填空题】
  - 0 非必填，此题的有效填写会提升审核通过率
    - ✔️ 非必填。「？」
- 1546452 **请将上一题填写的其他技术交流平台贡献信息，上传相关截图凭证，以供审核使用** 【文件题】
  - 0 非必填，此题的有效填写会提升审核通过率
    - ✔️ 非必填。如您上一题填写了相关内容，请于本题上传截图凭证。
- 1547175 **【温馨提示】将下面的提示文案内容完整输入至答题框内** 【填空题】
  - 0 本人知悉需要在倒计时结束前提交本问卷，超时提交将无法展示分数。
    - ✔️ 至少需要输入 20 个字符。本人知悉需要在倒计时结束前提交本问卷，超时提交将无法展示分数。

**Comment by pierrecashon** [comment]
关键填空题的答案就是那样的，答来答去就是那几句话，，能不一样吗

**Comment by sherves** [comment]
小米对于bl解锁的态度现在是真傻B，我就是为了能解锁才买的14pro，现在搞这一出，逼人换手机是吧

**Reply by pierrecashon** [reply]
我也是一样，手机都想卖了，关键这填空题是有标准答案的啊，你答案要是对不上那不照样0分，你答案对上了就是雷同。。简直蠢爆了

**Reply by Littlechilde** [reply]
都考虑和楼上老哥一起走key了

**Reply by linlei175323** [reply]
key的服务器最近好像都没开

**Comment by duriann** [comment]
没有答案吗

**Comment by q82415670** [comment]
第一阶段和第二阶段的题目答案是一样的吗

**Comment by s88868888** [comment]
不想说话，已经答题5次了，时间根本不够

**Comment by BigDaaddyy** [comment]
真的恶心人，现在答题时间都卡这么严了

**Comment by LynZiRui** [comment]
高考都没这么难吧，时间也没这么少😡

**Comment by linlei175323** [comment]
我最怕明天的题又搞好多不一样的
